### PR TITLE
Remove "using namespace mlx::core" in benchmarks/examples

### PR DESCRIPTION
**Pull request by zcbenz**
Refs https://github.com/ml-explore/mlx/issues/1513.

In order to make the code compilable with MSVC we must avoid introducing function `std` into current scope (see https://github.com/ml-explore/mlx/pull/1679 for more information).

This PR converts the code in benchmarks/examples to use `mx::` namespace alias, as an alternative solution to #1679. The downside is lots of lines get overwritten and it would be harder to check the code history in future.

I'll convert the code in python/src if you are OK with this. 

(The refactoring was mostly done with the help of AI editors, I check the code for a few times but errors can happen.)

### CODE REVIEW

**Review by awni — APPROVED** [review]
Looks good, thanks! Will merge when the tests clear.

File: examples/extensions/axpby/axpby.cpp [hunk 1]
```diff
@@ -32,9 +32,9 @@ namespace mlx::core {
  *  Follow numpy style broadcasting between x and y
  *  Inputs are upcasted to floats if needed
  **/
-array axpby(
-    const array& x, // Input array x
-    const array& y, // Input array y
+mx::array axpby(
```
Comment (awni) [inline examples/extensions/axpby/axpby.cpp:35]: I'm not sure why this is inside the `mlx::core` namespace. I think it makes more sense to keep it in a top-level / user name space and use the `namespace mx = ...`

Comment (awni) [reply examples/extensions/axpby/axpby.cpp:35]: Do you mind also updating `axbpy.h`?

Comment (zcbenz) [reply examples/extensions/axpby/axpby.cpp:35]: I have fixed the building and updated the extension to use its own namespace.

**Comment by zcbenz** [comment]
Sorry I forgot to test the code on a M chip mac, can you run tests again?

**Pull request merged**